iam/x509-identity-mgmt/js/src/notifications/NotificationEngine.js

```diff
@@ -0,0 +1,56 @@
+const { v4: uuidv4 } = require('uuid');
+
+class NotificationEngine {
+  constructor({
+    notificationKafkaProducer, logger, service, contentType,
+  }) {
+    Object.defineProperty(this, 'producer', { value: notificationKafkaProducer });
+    Object.defineProperty(this, 'logger', { value: logger });
+    Object.defineProperty(this, 'service', { value: service });
+    Object.defineProperty(this, 'contentType', { value: contentType });
+  }
+
+  /**
+   * initializes the Notification Engine
+   */
+  async start() {
+    await this.producer.connect();
+  }
+
+  /**
+   * stops the Notification Engine
+   */
+  async stop() {
+    await this.producer.disconnect();
+  }
+
+  async notify({
+    tenant, topic, eventType, eventData, partitionKey,
+  }) {
+    const message = this.generateMessage(tenant, eventType, eventData);
+    const msgStr = JSON.stringify(message);
+    await this.producer.produce(topic, msgStr, partitionKey);
+  }
+
+  generateMessage(tenant, eventType, eventData) {
+    return {
+      metadata: this.generateMetadata(tenant),
+      data: {
+        eventType,
+        eventData,
+      },
+    };
+  }
+
+  generateMetadata(tenant) {
+    return {
+      msgid: uuidv4(),
+      ts: Date.now(),
```

Comment:
In device-data e device manager topic the metadata used to have the key "timestamp" instead of "ts"...
But I think It's not a problem.. Maybe we should change to ts, it's more simple.

Reply:
Okay, I have no problem with maintaining `timestamp`. I will change the code!